File: src/test/java/org/demo/AggregateTest.java
package org.demo;

import org.axonframework.test.aggregate.AggregateTestFixture;
import org.axonframework.test.aggregate.FixtureConfiguration;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.TestInstance;

import java.lang.reflect.ParameterizedType;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class AggregateTest<T> {
	protected FixtureConfiguration<T> fixture;

	@BeforeEach
	@SuppressWarnings("unchecked")
	public void createFixture() {
		fixture = new AggregateTestFixture<>((Class<T>) ((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[0]);
	}
}

File: src/test/java/org/demo/AggregateTest.kt
package org.demo

import org.axonframework.test.aggregate.AggregateTestFixture
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.TestInstance
import kotlin.reflect.KClass

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
open class AggregateTest<T: Any>(private val clazz: KClass<T>) {
    protected lateinit var fixture: AggregateTestFixture<T>
    @BeforeEach
    fun createFixture() {
		fixture = AggregateTestFixture(clazz.java)
        fixture.setReportIllegalStateChange(false)
    }
}
